core/src/test/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingServiceSpec.scala

```diff
@@ -0,0 +1,131 @@
+package org.broadinstitute.dsde.rawls.spendreporting
+
+import java.util
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import com.google.api.services.bigquery.model.{GetQueryResultsResponse, Job, JobReference, QueryParameter, TableCell, TableRow}
+import org.broadinstitute.dsde.rawls.RawlsExceptionWithErrorReport
+import org.broadinstitute.dsde.rawls.dataaccess.{SamDAO, SlickDataSource}
+import org.broadinstitute.dsde.rawls.dataaccess.slick.TestDriverComponent
+import org.broadinstitute.dsde.rawls.model.{SamBillingProjectActions, SamResourceAction, SamResourceTypeNames}
+import org.broadinstitute.dsde.rawls.util.MockitoTestUtils
+import org.broadinstitute.dsde.workbench.google.GoogleBigQueryDAO
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.joda.time.DateTime
+import org.mockito.ArgumentMatchers.{any, eq => mockitoEq}
+import org.mockito.Mockito.{RETURNS_SMART_NULLS, when}
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+import org.scalatestplus.mockito.MockitoSugar
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+import scala.jdk.CollectionConverters._
+
+
+class SpendReportingServiceSpec extends AnyFlatSpecLike with TestDriverComponent with MockitoSugar with BeforeAndAfterAll with Matchers with ScalatestRouteTest with MockitoTestUtils {
+  val firstRow: util.List[TableCell] = List(
+    new TableCell().setV("0.0"), // cost
```

Comment:
🤔 There's a `set(<field_name>, <value>)` method that might make these comments unneeded, i.e., 
`new TableCell().set("cost", "0.0")`

Reply:
Unfortunately, the json that BQ returns looks like this
```{"f":[{"v":"0.003552000000000001"},{"v":"0.0"},{"v":"USD"},{"v":"2022-02-17"}]} // <- bigQueryDAO.getQueryResult(jobStatus).getRows().head```

Using `set(<field_name>, <value>)` results in a far more readable json object, but it is also incompatible with `extractSpendReportingResults` which expects data like what BQ returns. 

Sadly, this also means we can't use `get(<field_name>)` to clean up `extractSpendReportingResults` as all the fields are named `v`. We'll have to rely on the `getF.get(n).getV` pattern for now

Reply:
Wow that is super annoying, thanks for clarifying.

Reply:
There are currently two ways of working with BigQuery in Rawls. There's the `GoogleBigQueryDAO` (used in current workflow cost reporting) and the `BigQueryService` (used for TDR integration) which correspond to the two different GCP BigQuery Java clients. The `BigQueryService` is newer and may be easier to work with. I may play around with it a bit in a separate PR to see how it compares.

Little extra context on the two separate GCP BigQuery Java clients as I understand it. Early GCP API client libraries all fell under the `com.google.api` umbrella and tend to use less idiomatic Java. We use these client libraries through `workbench-libs` in the `google` package and they tend to be found in older parts of our system. The latest and greatest GCP client libraries are their [cloud client libraries](https://cloud.google.com/apis/docs/cloud-client-libraries) which fall under the `com.google.cloud` umbrella and are more idiomatic. We access these again through `workbench-libs`, but in the `google2` package instead.